src/Microsoft.Framework.Caching.SqlServer/SqlServerCache.cs

```diff
@@ -0,0 +1,478 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Data;
+using System.Data.SqlClient;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Framework.Caching.Distributed;
+using Microsoft.Framework.Internal;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.OptionsModel;
+
+namespace Microsoft.Framework.Caching.SqlServer
+{
+    /// <summary>
+    /// Distributed cache implementation using Microsoft SQL Server database.
+    /// </summary>
+    public class SqlServerCache : IDistributedCache
+    {
+        /// <summary>
+        /// To prevent frequent updates to a cache item's expiration time in database, especially when doing a
+        /// Get operation, a cache item's expiration time is extended by doubling the sliding expiration.
+        /// The 'SlidingExpiration' here should be interpreted as the minimum time to live for a cache item before
+        /// it expires.
+        /// Example:
+        /// For a sliding expiration of 30 mins, a new cache item would have its 'ExpiresAtTimeUTC' 60 mins(30 * 2)
+        /// from now. All 'Get' operations before the first 30 mins do not cause any database updates to the cache
+        /// item's expiration time and any 'Get' operations between the 30th and 60th minute would cause a database
+        /// update where the expiration time is again extended by 60 mins.
+        /// </summary>
+        internal const int ExpirationTimeMultiplier = 2;
+
+        // Table column names
+        private const string CacheItemId = "Id";
+        private const string CacheItemValue = "Value";
+        private const string ExpiresAtTimeUTC = "ExpiresAtTimeUTC";
+        private const string SlidingExpirationInTicks = "SlidingExpirationInTicks";
+        private const string AbsoluteExpiration = "AbsoluteExpiration";
+        private int _cacheItemIdIndex = -1;
+        private int _cacheItemValueIndex = -1;
+        private int _expiresAtTimeUTCIndex = -1;
+        private int _slidingExpirationInTicksIndex = -1;
+        private int _absoluteExpirationIndex = -1;
+
+        private readonly SqlServerCacheOptions _options;
+        private readonly ILogger _logger;
+        private readonly ISystemClock _systemClock;
+        private readonly SqlQueries _sqlQueries;
+        private DateTimeOffset _lastExpirationScan;
+
+        public SqlServerCache(IOptions<SqlServerCacheOptions> options, ILoggerFactory loggerFactory)
```

Comment:
Review if we are actually using logging (it is ok if we are going to add it in some places in the near future)